.github/workflows/pr-linter.yml

```diff
@@ -31,16 +31,39 @@ concurrency:
   cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
 
 jobs:
-  pr-formatting:
-    name: Formatting
-    runs-on: ubuntu-22.04
+  pr-changes:
+    name: Check changes
+    runs-on: ubuntu-latest
+    outputs:
+      changed: ${{ steps.changes.outputs.changed }}
     steps:
       - name: Checkout
         uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
         with:
           fetch-depth: 0
       - name: Check commit messages
         run: python3 ./.github/automation/commit-msg-check.py "${{ github.event.pull_request.head.sha }}" "${{ github.event.pull_request.base.sha }}"
+      - name: Check codebase changes
+        id: changes
+        run: |
+          if git diff --name-only origin/${{ github.base_ref }}..origin/${{ github.head_ref }}  | grep -q -E "^.github/|^examples/|^include/|^src/|^tests/"; then
+            echo "Found changes in codebase"
+            echo "changed=true" >> "$GITHUB_OUTPUT"
+          else
+            echo "No changes in codebase"
+            echo "changed=false" >> "$GITHUB_OUTPUT"
+          fi
+
+  pr-clang-format:
+    name: Formatting
```

Comment:
This check must run unconditionally on all PRs so that we can mark it as 'Required'. If a required check does not run for some reason PR promotion gets blocked.